capsul/pipeline/pipeline.py

```diff
@@ -513,6 +513,14 @@ def add_link(self, link, weak_link=False):
             if source_trait.output and not dest_trait.output:
                 dest_trait.connected_output = True
 
+        # Propagate the description in case of destination switch node
+        if isinstance(dest_node, Switch):
+            source_trait = source_node.process.trait(source_plug_name)
```

Comment:
source_node is maybe not a ProcessNode or a pipeline (it may be a switch), so source_node.process doesn't necessarily exist.
Use source_node.get_trait() instead.